assets/shared/helpers/player/videopress-adapter.js

```diff
@@ -61,7 +74,25 @@ export const getDuration = ( player ) =>
 			reject( new Error( 'Video duration not found' ) );
 		}
 
-		resolve( parseFloat( player.dataset.duration ) );
+		resolve( parseFloat( duration ) );
+	} );
+
+/**
+ * Get the current video time.
+ *
+ * @param {HTMLIFrameElement} player The player element.
+ *
+ * @return {Promise<number>} The current video time in seconds through a promise.
+ */
+export const getCurrentTime = ( player ) =>
+	new Promise( ( resolve, reject ) => {
+		const { currentTime } = player.dataset;
+
+		if ( ! currentTime ) {
+			reject( new Error( 'Video current time not found' ) );
+		}
```

Comment:
This isn't exactly needed because after a promise status is set, it can't be changed, but WDYT about adding a `return` here? :thinking:

Reply:
Sure! Good idea! https://github.com/Automattic/sensei/pull/5410/commits/957f560c5fb33eea344b0e9b9c9f3dcb5d838a36